src/gleam/list.gleam

```diff
@@ -1834,19 +1835,34 @@ fn partition_loop(list, categorise, trues, falses) {
 pub fn permutations(list: List(a)) -> List(List(a)) {
   case list {
     [] -> [[]]
-    [_, ..] ->
-      index_map(list, fn(i, i_idx) {
-        index_fold(list, [], fn(acc, j, j_idx) {
-          case i_idx == j_idx {
-            True -> acc
-            False -> [j, ..acc]
-          }
-        })
-        |> reverse
-        |> permutations
-        |> map(fn(permutation) { [i, ..permutation] })
-      })
-      |> flatten
+    l -> zipper(l, [], [])
+  }
+}
+
+fn zipper(list: List(a), rest: List(a), acc: List(List(a))) -> List(List(a)) {
+  case list {
+    [] -> reverse(acc)
+    [el, ..tail] ->
```

Comment:
One final little note. stdlib convention is to call these two `[head, ..tail]`, could you update both `zipper` and `zip_prepend` following this convention? Sorry I forgot to point it out in my first comment! 🙇‍♂️